OSVRUnreal/Plugins/OSVR/Source/OSVR/Private/OSVREntryPoint.cpp

```diff
@@ -29,13 +29,33 @@ DEFINE_LOG_CATEGORY(OSVREntryPointLog);
 
 OSVREntryPoint::OSVREntryPoint()
 {
+}
+
+void OSVREntryPoint::Initialize()
+{
+    // avoid double initialization
+    if (osvrClientContext != nullptr)
+    {
+        // too spammy to log here
+        return;
+    }
+
+    // hack: to avoid delay on OSVR server connection attempt when not using OSVR, don't initialize if nohmd command line param specified
+    if (FParse::Param(FCommandLine::Get(), TEXT("nohmd")))

```

Comment:
This is more about end-users who can not enable or disable UE4 plugins.   The end-user is using a full game build (in which plugins can't be enabled or disabled - the plugins enabled by developer are enabled and can't change) and not UE4Editor (where a developer can enable or disable plugins).

We have to target end-users with game builds and make the experience for them good - we (and all developers of products for non-developer end-users) can't force our non-OSVR users (the majority of our users) to wait for 3-10 seconds on a black screen while the UE4 OSVR-plugin fails to connect to a non-existent OSVR server.

P.S.   Sorry for long response delay - was out of office for a week.